### PR TITLE
feat: add TransactionClosedEvent send on queue

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes

Add logic for send TransactionClosedEvent on the closure event queue in case the transactions was previously authorized and Nodo closePayment response outcome is KO or the HTTP response status code is an unrecoverable ones (such as 400).
This event is sent with visibility timeout set to 0 so that events dispatcher can process it as soon as possible and perform transaction rollback logic.

#### Motivation and Context

This newly written event will be read by ecommerce event-dispatcher in order to perform transaction refund.

#### How Has This Been Tested?
Local junit tests + local test using ecommerce local mocking PGS and Nodo responses

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 1]
```diff
@@ -270,15 +267,15 @@ private ClosePaymentRequestV2Dto.OutcomeEnum authorizationResultToOutcomeV2(
         }
     }
 
-    private Mono<TransactionEvent<TransactionClosureData>> buildClosureEvent(
+    private Mono<TransactionEvent<TransactionClosureData>> processClosureEvent(
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java:270]: This method does the following things:
 * create a new closure event, either closed or closureFailed
 * decide whether it's enqueueable
 * enqueue it

I would split the method (especially the first two points from the rest), as right now it's not very clear from the outside what it's doing

File: src/test/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandlerTest.java [hunk 2]
```diff
@@ -1647,6 +1684,598 @@ void shouldEnqueueErrorEventOnNodoRecoverableFailure() {
...
+        TransactionActivatedEvent transactionActivatedEvent = new TransactionActivatedEvent(
+                transactionId.value().toString(),
+                new TransactionActivatedData(
+                        email.value(),
+                        PaymentNotices,
+                        faultCode,
+                        faultCodeString,
+                        Transaction.ClientId.CHECKOUT
+                )
+        );
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandlerTest.java:1715]: Can you use utilities from `TransactionTestUtils`?

**Review by giovanniberti — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java (hunk 1 above)
Comment (pietro-tota) [reply src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java]: Fixed in https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/148/commits/e8807b65afebce1c60687eab81d4e543b0efadf7

File: src/test/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandlerTest.java (hunk 2 above)
Comment (pietro-tota) [reply src/test/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandlerTest.java]: Fixed in https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/148/commits/e8807b65afebce1c60687eab81d4e543b0efadf7

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**